zeroex/orderwatch/order_watcher.go

```diff
@@ -144,62 +145,24 @@ func (w *Watcher) Watch(ctx context.Context) error {
 	w.wasStartedOnce = true
 	w.mu.Unlock()
 
-	// Create a child context so that we can preemptively cancel if there is an
-	// error.
-	innerCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
+	g, ctx := errgroup.WithContext(ctx)
 
-	// A waitgroup lets us wait for all goroutines to exit.
-	wg := &sync.WaitGroup{}
-
-	// Start some independent goroutines, each with a separate channel for communicating errors.
-	mainLoopErrChan := make(chan error, 1)
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		mainLoopErrChan <- w.mainLoop(innerCtx)
-	}()
-	cleanupLoopErrChan := make(chan error, 1)
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		cleanupLoopErrChan <- w.cleanupLoop(innerCtx)
-	}()
-	removedCheckerLoopErrChan := make(chan error, 1)
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		removedCheckerLoopErrChan <- w.removedCheckerLoop(innerCtx)
-	}()
-
-	// If any error channel returns a non-nil error, we cancel the inner context
-	// and return the error. Note that this means we only return the first error
-	// that occurs.
-	select {
-	case err := <-mainLoopErrChan:
-		if err != nil {
-			logger.WithError(err).Error("error in orderwatcher mainLoop")
-			cancel()
-			return err
-		}
-	case err := <-cleanupLoopErrChan:
-		if err != nil {
-			logger.WithError(err).Error("error in orderwatcher cleanupLoop")
-			cancel()
-			return err
-		}
-	case err := <-removedCheckerLoopErrChan:
-		if err != nil {
-			logger.WithError(err).Error("error in orderwatcher removedCheckerLoop")
-			cancel()
+	namedLoops := []struct {
+		loop func(context.Context) error
+		name string
+	}{{w.mainLoop, "mainLoop"}, {w.cleanupLoop, "cleanupLoop"}, {w.removedCheckerLoop, "removedCheckerLoop"}}
+	for _, namedLoop := range namedLoops {
+		namedLoop := namedLoop
```

Comment:
I realize this is here to avoid a common gotcha in Go, but it's not very obvious. Can you add a brief comment explaining why this is necessary?

Reply:
I added a link to the golang faq, which is what I have seen other go examples.